pandas/core/arrays/period.py

```diff
@@ -52,13 +71,17 @@ def _period_array_cmp(cls, op):
 
     def wrapper(self, other):
         op = getattr(self._ndarray_values, opname)
+        if isinstance(other, (ABCSeries, ABCIndexClass)):
+            # TODO: return NotImplemented?
+            other = other._values
```

Comment:
why is this not .values?

Reply:
We had a discussion related to this yesterday.

Some time ago, we added this explanation to the Index class:

https://github.com/pandas-dev/pandas/blob/145c2275e3560edc30ff418a57df67ba3c4c30d6/pandas/core/indexes/base.py#L702-L732

referring to `_values` as the "best array representation", so the attribute that will always be loss-less, and thus always will return the ExtensionArray if the Index (or Series) is backed by an ExtensionArray. 

By using this, that decouples this PR from the question what the public `.values` should do (keep returning ndarray of objects, or start returning PeriodArray?)

Reply:
ok